File: android/src/main/kotlin/ai/formx/mobile/sdk/formx_sdk_flutter/FormXSDKInitializer.kt
package ai.formx.mobile.sdk.formx_sdk_flutter

import ai.formx.mobile.sdk.FormXAPIClient

object FormXSDKInitializer {
    private var accessToken: String? = null
    private var endpoint: String? = null
    var formId: String? = null
    private var currentApiClient: FormXAPIClient? = null

    fun init(formId: String, accessToken: String, endpoint: String? = null) {
        this.formId = formId
        this.accessToken = accessToken
        this.endpoint = endpoint
        this.currentApiClient = FormXAPIClient(accessToken, endpoint)
    }

    val apiClient: FormXAPIClient?
        get() {
            return currentApiClient;
        }

    val isInitialized: Boolean
        get() {
            return accessToken != null && formId != null
        }
}

File: ios/Classes/FormXSDKInitializer.swift
//
//  FormXSDKInitializer.swift
//  formx_sdk_flutter
//
//  Created by wu mark on 2023/12/18.
//
import FormX

class FormXSDKInitializer {
    static let shared = FormXSDKInitializer()
    var extractorId: String?
    
    private var currentApiClient: FormXAPIClient? = nil
    
    func `init`(extractorId: String, accessToken: String, apiHost: String? = nil) {
        self.extractorId = extractorId
        currentApiClient = FormXAPIClient(accessToken: accessToken, apiHost: apiHost)
    }
    
    var apiClient: FormXAPIClient? {
        return currentApiClient
    }
}
